aws_helper/config.go

```diff
@@ -167,6 +168,17 @@ func GetAWSCallerIdentity(config *AwsSessionConfig, terragruntOptions *options.T
 	return *identity, nil
 }
 
+// Get the AWS Partition of the current session configuration
+func GetAWSPartition(config *AwsSessionConfig, terragruntOptions *options.TerragruntOptions) (string, error) {
+	identity, err := GetAWSCallerIdentity(config, terragruntOptions)
+	if err != nil {
+		return "", errors.WithStackTrace(err)
+	}
+
+	arn_slice := strings.Split(*identity.Arn, ":")
+	return arn_slice[1], nil
```

Comment:
NIT: there is an [`arn` package](https://docs.aws.amazon.com/sdk-for-go/api/aws/arn/) in the SDK that can do this with somewhat better error handling.

Reply:
I was thinking I should use `arnSlice` for the var name anyway so... happy to modify that!